domain-server/src/DomainServerAcmeClient.cpp

```diff
@@ -0,0 +1,751 @@
+//
+//  DomainServerAcmeClient.cpp
+//  domain-server/src
+//
+//  Created by Nshan G. on 2021-11-15.
+//  Copyright 2021 Vircadia contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "DomainServerAcmeClient.h"
+
+#include <HTTPManager.h>
+#include <HTTPConnection.h>
+#include <PathUtils.h>
+#include <acme/acme-lw.hpp>
+#include <acme/ZeroSSL.hpp>
+
+#include <QDir>
+
+#include <memory>
+#include <array>
+#include <set>
+
+#include "DomainServerSettingsManager.h"
+
+
+Q_LOGGING_CATEGORY(acme_client, "vircadia.acme_client")
+
+using namespace std::literals;
+using std::chrono::system_clock;
+
+std::string readAll(const QString& path)
+{
+    QFile file(path);
+    if(file.open(QFile::ReadOnly)) {
+        return file.readAll().toStdString();
+    }
+    return std::string();
+}
+
+bool writeAll(const std::string& data, const QString& path)
+{
+    QFile file(path);
+    return file.open(QFile::WriteOnly) &&
+        file.write(QByteArray::fromStdString(data)) == qint64(data.size());
+}
+
+
+class AcmeHttpChallengeServer : public AcmeChallengeHandler, public HTTPRequestHandler {
+    struct Challenge {
+        QUrl url;
+        QByteArray content;
+    };
+
+public:
+    AcmeHttpChallengeServer() :
+        manager(QHostAddress::AnyIPv4, 80, "", this)
+    {}
+
+    void addChallenge(const std::string&, const std::string& location, const std::string& content) override {
+        challenges.push_back({
+            QString::fromStdString(location),
+            QByteArray::fromStdString(content)
+        });
+    }
+
+    std::chrono::milliseconds selfCheckDuration() override { return 1s; }
+    std::chrono::milliseconds selfCheckInterval() override { return 250ms; }
+
+    bool handleHTTPRequest(HTTPConnection* connection, const QUrl& url, bool skipSubHandler = false) override {
+        auto challenge = std::find_if(challenges.begin(), challenges.end(), [&url](auto x) { return x.url == url; });
+        if(challenge != challenges.end()) {
```

Comment:
```suggestion
        if (challenge != challenges.end()) {
```
And similarly for other if()s.